ts/runtime.ts

```diff
@@ -23,39 +23,60 @@ import { Instance } from "./instance";
 import { Pool } from "./pool";
 import { Module } from "./module";
 import { Cache } from "./cache";
+import { DisabledWASI } from "./wasi";
 
 export type NextFn<T extends Signature> = (ctx: T) => T;
 
 export async function New(functions: ScaleFunc[]): Promise<Runtime<httpSignature.Context>> {
  return NewFromSignature(httpSignature.New, functions);
 }
 
-export async function NewFromSignature<T extends Signature>(newSignature: NewSignature<T>, functions: ScaleFunc[]): Promise<Runtime<T>> {
+export async function NewFromSignature<T extends Signature>(newSignature: NewSignature<T>, functions: ScaleFunc[]): Promise<RawRuntime<T>> {
   const r = new Runtime(newSignature, functions);
   await r.Ready();
   return r;
 }
 
-export class Runtime<T extends Signature> {
+export async function NewRaw(functions: Uint8Array[]): Promise<Runtime<httpSignature.Context>> {
+  return NewRawFromSignature(httpSignature.New, functions);
+}
+
+export async function NewRawFromSignature<T extends Signature>(newSignature: NewSignature<T>, functions: Uint8Array[]): Promise<RawRuntime<T>> {
+  const r = new RawRuntime(newSignature, functions);
+  await r.Ready();
+  return r;
+}
+
+export class RawRuntime<T extends Signature> {
   public NewSignature: NewSignature<T>;
   private readonly ready: Promise<any>;
   private functions: Func<T>[];
   public head: undefined | Func<T>;
   public tail: undefined | Func<T>;
 
-  constructor(newSignature: NewSignature<T>, functions: ScaleFunc[]) {
+  constructor(newSignature: NewSignature<T>, functions: WebAssembly.Module[] | Uint8Array[]) {
     this.NewSignature = newSignature;
     this.functions = [];
 
     this.ready = new Promise(async (resolve) => {
       for (let i = 0; i < functions.length; i++) {
         const fn = functions[i];
-        const mod = await WebAssembly.compile(fn.Function as Buffer);
-
-        const f = new Func<T>(fn, mod);
+        let f: Func<T>;
+        if (fn instanceof Uint8Array) {
+          const instantiatedSource =  await WebAssembly.instantiate(fn, {
```

Comment:
Why are we changing `WebAssembly.compile` to `WebAssembly.instantiate` and then using the `.module` property from that?
Afaik if compile is disabled, then instantiate would be as well.
I would say the compile is cleaner and more readable.

It's good to allow passing in a WebAssembly.Module directly here so we can import .wasm directly for NextJS etc.

Reply:
FWIW, We could just allow ScaleFn[] | WebAssembly.Module[] | Uint8Array[] as inputs here, then it could be used in all scenarios without any changes to existing code. (Get rid of RawRuntime and just make Runtime accept a variety of inputs).

Reply:
I've cleaned it up so that we no longer allow the Uint8 array or the WebAssembly.Module - both of those are prone to errors with different signatures and what not since they're not type checking a signature. 

Now we support ScaleFns and `scale.Func` objects - both of which typecheck a signature.